nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -13,6 +15,134 @@
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    def __init__(self, *args, stub=False, **kwargs):
+        self.vc = cv2.VideoCapture(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        self.stub = stub
+        self._current_frame = 0
+        self.frame_count = self.get_movie_frame_count()
+        assert self.frame_count > 0, "movie contains no frames"
+        self.fps = self.get_movie_fps()
+        self.frame = self.get_movie_frame(0)
+        assert self.frame is not None, "unable to read the movie file provided"
+
+    def get_movie_timestamps(self):
+        """
+        Return numpy array of the timestamps for a movie file.
+
+        """
+        if not self.vc.isOpened():
+            raise ValueError("movie file is not open")
+        ts = [self.vc.get(cv2.CAP_PROP_POS_MSEC)]
+        for i in tqdm(range(1, self.get_movie_frame_count()), desc="retrieving video timestamps"):
+            self.current_frame = i
+            ts.append(self.vc.get(cv2.CAP_PROP_POS_MSEC))
+        self.current_frame = 0
+        return np.array(ts)
+
+    def get_movie_fps(self):
+        """
+        Return the internal frames per second (fps) for a movie file.
+
+        """
+        if int(cv2.__version__.split(".")[0]) < 3:
+            return self.vc.get(cv2.cv.CV_CAP_PROP_FPS)
+        return self.vc.get(cv2.CAP_PROP_FPS)
+
+    def get_frame_shape(self) -> Tuple:
+        """
+        Return the shape of frames from a movie file.
+        """
+        return self.frame.shape
+
+    def get_movie_frame_count(self):
+        """
+        Return the total number of frames for a movie file.
+
+        """
+        if self.stub:
+            # if stub the assume a max frame count of 10
+            return 10
```

Comment:
```suggestion
        if self.stub:
            # if stub the assume a max frame count of 10
            return 10
```

See comment above; probably for the best if stubbing is outside the purview of this class.

Reply:
setting stub internally helps to correctly iterate over when its used as an utterable:
```python
vcc = VideoCaptureContext("movie_file_path.avi", stub=True)
frames=[]
for frame in vcc:
   frames.append(frame)
len(frame) # this would be 10 
```
If we don't have the stub option then we would have to iterate over all frames (I/O for all frames) and then select only the stub from that. It would be wasteful. 
the `__next__` raises `StopIteration` if the `frame_no` is greater than that allowed by stub.

Reply:
Not necessarily, we would simply override the `my_video_capture.frame_count = stub_amount` after `my_video_capture = VideoCaptureContext(...)`; hence https://github.com/catalystneuro/nwb-conversion-tools/pull/392/files#diff-4482e8014d6730ce58954d23f828a997e8af7bd61b2d287c7b562f484e5e8733R97 would behave exactly the same way, but with full outside control of stubbing, and not the responsibility of this class.

Reply:
yeah, I like the idea of making `self.frame_count` a property and allow the user to set it to override the number of frames. Indeed it allows more flexibility over fixing a value if `stub=True`. I have made changes https://github.com/catalystneuro/nwb-conversion-tools/pull/392/files#diff-4482e8014d6730ce58954d23f828a997e8af7bd61b2d287c7b562f484e5e8733R45